pkgs/node/src/chain.zig

```diff
@@ -308,10 +311,10 @@ pub const BeamChain = struct {
             block.slot,
         });
 
-        const post_state = blockInfo.postState orelse computedstate: {
+        const post_state = if (blockInfo.postState) |post_state_ptr| post_state_ptr.* else computedstate: {
             // 1. get parent state
-            const pre_state = self.states.get(signedBlock.message.parent_root) orelse return BlockProcessingError.MissingPreState;
-            var cpost_state = try types.sszClone(self.allocator, types.BeamState, pre_state);
+            const pre_state_ptr = self.states.get(signedBlock.message.parent_root) orelse return BlockProcessingError.MissingPreState;
+            var cpost_state = try types.sszClone(self.allocator, types.BeamState, pre_state_ptr.*);
```

Comment:
if you do the ssz change and make cpost_state as allocated pointer then post_state can directly be state pointer